### PR TITLE
Migrate over 4.4 lib dom and webworker

### PR DESCRIPTION
**Pull request by orta**
Finalizes changes from the beta to RC.

### CODE REVIEW

File: src/lib/dom.generated.d.ts [hunk 1]
```diff
@@ -152,6 +152,12 @@ interface BlobPropertyBag {
     type?: string;
 }
```
Comment (orta) [inline src/lib/dom.generated.d.ts:153]: Most of the changes from the RC are adding `deprecated` or switching the deprecated to multiline. Though this includes a lot of the beta feedback about the DOM changes in https://github.com/microsoft/TypeScript/issues/45047

File: src/lib/webworker.generated.d.ts [hunk 2]
```diff
@@ -5,14 +5,15 @@
 interface AddEventListenerOptions extends EventListenerOptions {
```
Comment (orta) [inline src/lib/webworker.generated.d.ts:5]: Big changes, but this brings this file up-to-date with the `dom.d.ts` changes. WebWorker being opt-in meant we focused on the dom changes for the beta and these follow on.

**Comment by orta** [comment]
@typescript-bot test this
@typescript-bot run dt
@typescript-bot user test this

**Comment by orta** [comment]
The node12 is from a timeout, which _Seems_ unrelated. Will re-trigger. A lot of the RWC errors seems to have come from the addition of `abort` to `AbortSignal` which https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54944 might fix - 

There's a good question of if we want to ship that change, given that it would force quite a few projects to update their node types version which isn't great.

**Comment by orta** [comment]
We're giving Saschanaz a bit of time to see:

 - https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/1093
 - https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/1092

and a revised:
- https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/1089

**Comment by sandersn** [comment]
For now I've cherry-picked those 3 PRs into this PR. We can change the types further in the final release of 4.4.

**Pull request merged**